### PR TITLE
Copies pydantic model config in subset model for pydantic integration

### PR DESCRIPTION
**Pull request by elijahbenizzy**
Beforehand we were dropping this. Thus features like allowing arbitrary fields was disallowed.

## Changes
- Adds config copy

## How I tested this
- Manually 
- Unit tests

## Notes
- Thanks @zilto for the flag

### CODE REVIEW

**Review by zilto — APPROVED** [review]
LGTM. Main failure point was `subset_model()`

File: pyproject.toml [hunk 1]
```diff
@@ -58,7 +58,7 @@ tests = [
   "langchain_community",
   "pandas",
   "psycopg2",
-  "pydantic",
+  "pydantic[email]",
```
Comment (skrawcz) [inline pyproject.toml:61]: email?

Comment (elijahbenizzy) [reply pyproject.toml:61]: test

Comment (zilto) [reply pyproject.toml:61]: Elijah said he added it only for testing purposes :welp:

**Pull request merged**